shell/platform/darwin/macos/framework/Source/FlutterView.mm

```diff
@@ -84,6 +86,17 @@ - (void)reshaped {
                             }];
 }
 
+- (NSColor*)backgroundColor {
```

Comment:
Instead of overriding getter wouldn't it be better to set blackColor as default value in constructor? That way user could override the color and possibly prevent additional layer creation (i.e. when background color is nil).

Also while I don't think the additional layer brings much overhead, in the end it is still another layer to composite. I'm probably missing bigger picture, but why can't the background just be painted by framework?

Reply:
Transparent makes sense to me, but I think that we have tests that will be affected if we do make the default background transparent.

Reply:
> AFAIK, if the framework painted the background color, all platforms would have the background color as black.

That is the intent. For consistency across platforms, the default background colour for Flutter apps is expected to be black. As @knopp mentions, on some platforms we lucked out and got a black background by default due to our implementation, on others, we need to do work to make it happen. Pushing a default background colour layer is one way we could solve this. 

If we went that route, there are a couple considerations:
* As @knopp suggests, we really should allow this to be overridden by the user. If the user elects to set a transparent background colour, if the native view has a black background though, the user will still get black. So if we went this route, I think we'd want to make sure that the default native view has a transparent background so we pick up whatever is underneath it.
* This will add a layer to the layer tree. If that's done in a way that's detectable by the user it's possible it breaks some unit tests. AFAIK there's no way for user code to extract the layer tree from `Scene` or `SceneBuilder` so potentially it could be done in there (or on the engine side in flow).

Reply:
> Pushing a default background colour layer is one way we could solve this. ... This will add a layer to the layer tree. If that's done in a way that's detectable by the user it's possible it breaks some unit tests. 

This may actually be a positive:
1. We'd be able to test that the background color is black using `dart:ui`, golden tests, or Flutter driver's screenshot feature.
2. I'd be curious to see which unit tests are broken and why, this seems fishy to be affected by this

Reply:
(1) applies regardless of how this is implemented, though not 100% how we'd test it using `dart:ui` unless we have an API to sample pixel colour. 

If we support setting the bg colour we'd probably want a (separate?) test that layers in a non-flutter view that renders some colour like red, then layers a flutter view on top and verifies that if the user sets a transparent background, the red is visible through the flutter view.

Agreed on (2); since AFAIK there's no easy way to poke at the underlying layer-tree from the framework, I'd expect

Reply:
> though not 100% how we'd test it using dart:ui unless we have an API to sample pixel colour

I believe you can create an `Image` and then verify the pixels. See [this](https://github.com/flutter/flutter/blob/9b2668a451a8d0e756225b5d29be2c48d8dd2c4d/packages/flutter_test/lib/src/_matchers_io.dart#L17-L31) or [this](https://github.com/flutter/flutter/blob/9b2668a451a8d0e756225b5d29be2c48d8dd2c4d/dev/integration_tests/ui/test_driver/screenshot_test.dart#L31-L39).

Reply:
> I believe you can create an Image and then verify the pixels. See [this](https://github.com/flutter/flutter/blob/9b2668a451a8d0e756225b5d29be2c48d8dd2c4d/packages/flutter_test/lib/src/_matchers_io.dart#L17-L31) or [this](https://github.com/flutter/flutter/blob/9b2668a451a8d0e756225b5d29be2c48d8dd2c4d/dev/integration_tests/ui/test_driver/screenshot_test.dart#L31-L39).

Do these methods capture the whole screen or just the `FlutterView`?

Reply:
From my understanding they both capture [the scene's entire layer tree](https://github.com/flutter/engine/blob/c33eae133e21ae6857b917b382d592fed12cb1b3/lib/ui/compositing/scene.cc#L89-L90), which doesn't include the whole screen.